test/functional/feature_asmap.py

```diff
@@ -98,6 +100,14 @@ def test_empty_asmap(self):
         self.node.assert_start_raises_init_error(extra_args=['-asmap'], expected_msg=msg)
         os.remove(self.default_asmap)
 
+    def test_missing_peers_dat_with_addrman_checks(self):
```

Comment:
In 0ab75d87b7ff1cb6ebb69ffe68342b88403230b9: This commit is confusing because it adds an addrman test to `feature_asmap.py`, which from what I can see is basically the same test that is [already in feature_addrman.py](https://github.com/bitcoin/bitcoin/blob/053a5fc7d912d597cd6dc7376b479420d1eae1c0/test/functional/feature_addrman.py#L82), with the addition of `-checkaddrman`, so isn't actually asmap specific? Later in this PR the two files get merged together, so either this new version or the existing version essentially gets deleted. Why not just add `-checkaddrman` to the existing test after the merge, rather than another commit which is essentially duplicate code.

Reply:
This commit was earlier.

The goal was to have minimum regression tests in place before the fix, or all the refactoring, or at least as soon as possible, to reduce the chance of accidentally adding more critical bugs.

I'm trying to catch up to a moving target and may slim this back down to just the first regression test.

Reply:
Dropped